File: solutions/038.count-and-say/count-and-say.java
class Solution {
    public String countAndSay(int n) {
        if(n == 1){
            return "1";
        }else{
            return count(countAndSay(n - 1));
        }
        
        
    }
    public String count(String s){
        StringBuilder res = new StringBuilder();
        char[] c = s.toCharArray();
        int continuous = 0;
        for(int i = 0; i < c.length; i++){
            if(i == 0 || c[i] == c[i - 1]){
                continuous++;
            }else{
                res.append(continuous);
                res.append(c[i - 1]);
                continuous = 1;
            }
            if(i == c.length -1){
                res.append(continuous);
                res.append(c[i]);
            }
        }
        return res.toString();
    }
    
}

File: solutions/038.count-and-say/count-and-say.js
/**
 * @param {number} n
 * @return {string}
 */
var countAndSay = function(n) {
    const res = ['1'];
    for(let i = 1; i < n; i++){
        res.push(helper(res[i -1]));
    }
    return res[n - 1];
    
    function helper(str){
        let res = '';
        let cnt = 0;
        for(let i = 0; i < str.length; i++){
            if(i > 0 && str[i] !== str[i - 1]){
                res += cnt + str[i - 1];
                cnt = 0;
            }
            cnt++;
        }
        res += cnt + str[str.length - 1];
        return res;
    }
    
};